index.js

```diff
@@ -60,17 +65,13 @@ function _getGame(players, chosenCards, flex){
 }
 
 function _classifyCards(cards) {
-    var deck = { negatives: [], nonnegatives: [] };
-    cards.map(function (card) {
-        if (card.value < 0)
-            for (var i = 0; i < card.amount; i++)
-                deck.negatives.push({ role: card.role, value: card.value, amount: 1 });
-        else
-            for (var i = 0; i < card.amount; i++)
-                deck.nonnegatives.push({ role: card.role, value: card.value, amount: 1 });
-    });
+    if(!cards || !Array.isArray(cards) || cards.length === 0) 
+        return [];
 
-    return deck;
+    return { 
+        negatives: cards.filter(c => c.value < 0), 
```

Comment:
This won't work. Let's say for example we receive this:

```
[
     { role: 'Villager', value: 1, amount: 4 }, 
     { role: 'Werewolf', value: -6 amount: 1 }
]
```

In order to balance the game we need to have all cards individually like this:

```
[
     { role: 'Villager', value: 1, amount: 1 }, 
     { role: 'Villager', value: 1, amount: 1 }, 
     { role: 'Villager', value: 1, amount: 1 }, 
     { role: 'Villager', value: 1, amount: 1 }, 
     { role: 'Werewolf', value: -6 amount: 1 }
]
```

Here we do that process:
```
cards.map(function (card) {
         if (card.value < 0)		
             for (var i = 0; i < card.amount; i++)		
                 deck.negatives.push({ role: card.role, value: card.value, amount: 1 });		
         else		
             for (var i = 0; i < card.amount; i++)		
                 deck.nonnegatives.push({ role: card.role, value: card.value, amount: 1 });		
     });
```